source/concurrent/WithMutex.h

```diff
@@ -0,0 +1,43 @@
+/* WithMutex.h
+Copyright (c) 2024 by tibetiroka
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef WITH_MUTEX_H_
+#define WITH_MUTEX_H_
+
+#include <mutex>
+
+
+
+// Helper class to add a mutex to Body without declaring the full copy constructor.
+// These operations are not thread safe. Do not copy the object in a concurrent context,
+// or with a locked mutex.
```

Comment:
Instead of having a comment that people likely wont read, this should be enforced by code

Reply:
If it could be enforced easily, it would be enforced in the standard library... There isn't really a "good" way to do it, just various trade-offs. This code works just fine unless you copy a body with a locked mutex, and expect the copied mutex to also be locked. But even if I lock the copy's mutex, the same comment would have to be made, as that wouldn't always be safe with temporaries...